### ISSUE TITLE
Dark mode redshift

### ISSUE THREAD

**Issue by adigitoleo** [ahrm/sioyek#998]
An out-of-the-box feature idea for dark mode. I mostly use it at night (bad for YOUR eyes maybe, mine are just fine... wait, is than an UFO?) on account of that is when The Real Lifeᵀᴹ also uses dark mode. In that case, in addition to `dark_mode_contrast` perhaps there could be `dark_mode_redshift` which would set some amount to red-shift all colors in the document, so that I may keep my biological EM receptors for a slightly longer duration.

I tried using the custom color mode, in which I can of course set whatever colors I want for text/background, however I prefer to keep the multichromatic images, otherwise some figure legends can become impossible to decipher. So I guess the alternative option would be `custom_monochromatic_images` which we could set to false to disable the monochromatic filter (if possible)?

Just an idea, haven't looked at the code yet.

**Comment by ahrm** [comment]
In the development branch we have a `preserve_image_colors_in_dark_mode` config option which keeps image colors in dark mode and custom color modes.

**Comment by adigitoleo** [comment]
OK that would likely be good enough for me. I could be picky and ask for RGB inverted images in custom color mode but that's not a big deal. Can be closed otherwise.

**Comment by ahrm** [comment]
Added a `inverted_preserved_image_colors` config option in https://github.com/ahrm/sioyek/commit/fc3b95b96f993543cc2ed52754fecbf77d8677ac.

**Comment by adigitoleo** [comment]
So I finally got around to building the development branch on my main PC. Right now I built from b312fe0, but I observe this behaviour:
- `preserve_image_colors_in_dark_mode` is recognized (no warning when `--verbose`) but does nothing; image colors are always preserved
- `inverted_preserve_image_colors` is not recognized (warning when `--verbose`); perhaps it wasn't hooked up right

I'm setting both of those in `$HOME/.config/sioyek/prefs_user.config`.

**Comment by ahrm** [comment]
>  but does nothing; image colors are always preserved

What do you mean by images being preserved?

> inverted_preserve_image_colors is not recognized

You misspelled the command, it should be `inverted_preserved_image_colors`

**Comment by adigitoleo** [comment]
Ah my bad, thanks for pointing out the spelling. The other part was just me being silly, it works fine (I forgot that the image color "inversion" just flips the lightness, not hue; I can observe that this is not happening if I set `preserve_image_colors_in_dark_mode 1`).

Sorry for the noise

**Comment by adigitoleo** [comment]
OK wait, but `inverted_preserved_image_colors 0` doesn't seem to turn off the RGB inversion in custom color mode. Screenshot with light mode, dark mode and custom mode, with `preserve_image_colors_in_dark_mode 1` and `inverted_preserved_image_colors 0`:

![inverted_preserved_image_colors_0](https://github.com/ahrm/sioyek/assets/34595875/cd83b2f8-d283-432c-90bb-09ccd99bdf70)

**Comment by ahrm** [comment]
Can you share your `prefs_user.config`?

**Comment by adigitoleo** [comment]
Here: https://x0.at/4NfQ.txt

**Comment by adigitoleo** [comment]
I can't reproduce with `tutorial.pdf` either. Mandelbrot seems to have preserved colors in both dark and custom mode. Maybe it is because my "images" (e.g. the orthorhombic dipyramid above) are PDF files themselves.

**Comment by adigitoleo** [comment]
The PDF file for which I observe the behaviour (in custom color mode): https://x0.at/4K53.pdf

**Comment by ahrm** [comment]
Yes, these are not "images" but vector graphics. There is no realistic way that we can distinguish these from the normal PDF contents.
